util/chplenv/chpl_arch.py

```diff
@@ -477,62 +461,136 @@ def get(location, map_to_compiler=False, get_lcd=False):
                 stderr.write("Warning: Could not detect the lowest common "
                              "denominator processor type for this platform. "
                              "You may be unable to use the Chapel compiler\n")
-        if is_known_arm(arch):
-            return arch_tuple('cpu', arch)
-        else:
-            return arch_tuple('arch', arch)
+        return arch
     elif 'pgi' in compiler_val:
-        return arch_tuple('none', 'none')
+        return 'none'
     elif 'cray' in compiler_val:
-        return arch_tuple('none', 'none')
+        return 'none'
     elif 'ibm' in compiler_val:
-        return arch_tuple('none', 'none')
+        return 'none'
+
+    return arch
+
+@memoize
+def default_arch(flag):
+    comm_val = chpl_comm.get()
+    compiler_val = chpl_compiler.get(flag)
+    platform_val = chpl_platform.get(flag)
+
+    arch = 'unknown'
+
+    if comm_val == 'none' and ('linux' in platform_val or
+                               platform_val == 'darwin' or
+                               platform_val.startswith('cygwin')):
+      # Clang cannot detect the architecture for aarch64.  Otherwise,
+      # let the backend compiler do the actual feature set detection. We
+      # could be more aggressive in setting a precise architecture using
+      # the double checking code above, but it seems like a waste of time
+      # to not use the work the backend compilers have already done
+      if compiler_val in ['clang', 'clang-included']:
+          if get_native_machine() == 'aarch64':
+              arch = 'unknown'
+          else:
+              arch = 'native'
+      else:
+            arch = 'native'
+
+    return arch
+
+# Returns the machine type for the passed arch.
+#  * if arch is native/none/unknown, return result of get_native_machine
+#  * otherwise returns the machine type corresponding to the selected arch
+@memoize
+def machine_for_arch(arch):
+    if arch == "native" or arch == "none" or arch == "unknown":
```

Comment:
Looks like there's inconsistent indentation here and other places. Can you unify to 4 spaces per indention level?